src/api/profile/ElevationProfileSegment.class.js

```diff
@@ -0,0 +1,11 @@
+export default class ElevationProfileSegment {
+    /** @param {ElevationProfilePoint[]} points */
+    constructor(points) {
+        /** @type {ElevationProfilePoint[]} */
+        this.points = [...points]
+    }
+
+    get hasData() {
+        return this.points.length >= 2 && this.points[0].elevation !== null
```

Comment:
Should you not use the `ElevationProfilePoint.hasData()` method here with the `any()`  method ?

Reply:
that would be better yes

Reply:
👍🏼